feat: 사용자별 레시피 북마크 조회/추가/삭제 구현

lilyshin opened the pull request:
### 🚀 Pull Request  
---  
- 사용자별 레시피 북마크 조회/추가/삭제 구현

### ✨ 변경 내용  
---  
- 사용자별 레시피 북마크 조회/추가/삭제 API 추가
- 레시피 북마크 테스트코드 추가
- 레시피, 레시피 북마크 entity 추가

### 🛠 테스트 방법  
---   
1. npm run test:e2e 실행

### 📌 관련 이슈  
---  
- #(관련 이슈 번호)  

### 📎 참고 사항  
---  
- 변경된 API 문서 링크  
- 관련된 디자인 또는 기획 문서 링크  
- 추가적인 설명이 필요한 부분  


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **신규 기능**
  * 북마크 기능 추가: 생성(POST /v1/user/bookmarks), 삭제(DELETE /v1/user/bookmarks/:recipeId), 날짜별 조회(GET /v1/user/bookmarks).
  * 공개 프로필 조회 추가: GET /v1/user/:id.
* **변경 사항**
  * 사용자 API 기본 인증(JWT) 적용 및 인증 모듈 통합 개선(Jwt 전략 노출).
  * 에러 코드 확장: BOOKMARK_ALREADY_EXISTS, BOOKMARK_NOT_FOUND, RECIPE_NOT_FOUND.
* **문서**
  * 북마크 요청/응답 스키마(생성 DTO, 그룹화 DTO, 레시피 포함 DTO) 추가.
* **테스트**
  * E2E 테스트 및 모의 모듈/헬퍼 추가(모의 앱, 인증 가드, 테스트 유틸).
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

lilyshin committed:
feat: 사용자별 레시피 북마크 조회/추가/삭제 구현

lilyshin committed:
test: 사용자 북마크 API 테스트코드 추가

lilyshin committed:
feat: 북마크 추가 시 레시피 존재 여부 확인 로직 추가

lilyshin committed:
feat: customRepository query alias 추가

lilyshin committed:
feat: 북마크 조회 시 필요한 컬럼만 반환, 레시피 타이틀 컬럼 추가

lilyshin committed:
feat: 불필요한 주석 제거

lilyshin committed:
refactor: 북마크 생성 API 리팩토링

lilyshin committed:
feat: package-lock.json rollback

lilyshin committed:
feat: package-lock.json rollback

realbig4199 commented on src/api/user/dto/bookmark-with-recipe.dto.ts:
JS/TS는 보통 변수명 카멜 케이스로 선언하는 데, 카멜 케이스로 좀 변경 부탁드려요

https://velog.io/@lobinb797/Day12-Javascirpt

lilyshin replied:
이번 브랜치는 이대로 머지하고 새로 브랜치 따서 한꺼번에 수정할게요~!

realbig4199 commented on src/api/user/user.service.ts:
getBookmarksByDate 이 메서드 날짜별로 그룹핑하는 이유가 뭐였죠? 제가 지금 보고 있는 피그마에는 날짜가 따로 안보여서;; 제가 파일 잘못 보고 있는건가 해서 여쭤봐요
<img width="824" height="795" alt="image" src="https://github.com/user-attachments/assets/4fc9339d-0ac5-4f43-8f49-b8a07c871765" />

lilyshin replied:
기획서가 아직 반영이 되지 않았네요... 
<img width="887" height="310" alt="image" src="https://github.com/user-attachments/assets/b2589c8e-bf6a-443a-9ddb-17e45d638962" />
요런 히스토리로 날짜별 그룹핑 추가했습니다.

lilyshin committed:
refactor: develop 코드 merge, husky test e2e 추가

- test/mocks/user.mock.ts에서 JwtGuard import 경로 수정
- test/helpers/auth.helper.ts에서 JwtGuard import 추가
- pre-commit 훅에 E2E 테스트 실행 추가
